components/req_cpu/src/linux.rs

```diff
@@ -0,0 +1,690 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::collector::{Collector, CollectorId};
+use crate::{RecorderConfig, RequestCpuRecords, RequestTag};
+
+use std::cell::Cell;
+use std::fs::read_dir;
+use std::marker::PhantomData;
+use std::sync::atomic::Ordering::{AcqRel, Acquire, Relaxed};
+use std::sync::atomic::{AtomicPtr, AtomicU64};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use collections::{HashMap, HashSet};
+use crossbeam::channel::{unbounded, Receiver, Sender};
+use lazy_static::lazy_static;
+use libc::pid_t;
+use procinfo::pid;
+use procinfo::pid::Stat;
+
+pub fn init_recorder() {
+    lazy_static! {
+        static ref PHANTOM: () = {
+            std::thread::Builder::new()
+                .name("req-cpu-recorder".to_owned())
+                .spawn(move || {
+                    let mut recorder = ReqCpuRecorder::new(RecorderConfig::default());
+
+                    loop {
+                        recorder.handle_collector_registration();
+                        recorder.handle_thread_registration();
+                        recorder.record();
+                        recorder.may_advance_window();
+                        recorder.may_gc();
+
+                        std::thread::sleep(Duration::from_micros(
+                            (recorder.config.record_interval_ms * 1_000.0) as _,
+                        ));
+                    }
+                })
+                .expect("Failed to create recorder thread");
+        };
+    }
+    *PHANTOM
+}
+
+pub fn register_collector(collector: Box<dyn Collector>) -> CollectorHandle {
```

Comment:
What is a collector?

Reply:
Collectors should handle records sent by the recorder per second.